src/detect-transform-pcrexform.c

```diff
@@ -125,6 +142,15 @@ static int DetectTransformPcrexformSetup (DetectEngineCtx *de_ctx, Signature *s,
         SCReturnInt(-1);
     }
 
+    size_t regexstr_len = MIN(PCRE_ID_MAX_DATA_LEN, strlen(regexstr));
```

Comment:
can we just `SCStrdup`? The length is bound by the rule max-len, so it should be safe.

Reply:
Will do.